### PR TITLE
Jd

### PR DESCRIPTION
**Pull request by JDWong-wag**
bunch of booking form stuff & new components

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+import momentLocalizer from 'react-widgets-moment';
+
 import store from './store/configureStore';
 import './index.css';
 import App from './App';
 import registerServiceWorker from './registerServiceWorker';
 
+// For react-widgets
+Moment.locale('en');
+momentLocalizer();
```
Comment (marklabs) [inline src/index.js]: what does this do? is this a proposed solution for production or prototype?

File: package.json [hunk 2]
```diff
@@ -3,16 +3,24 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/fontawesome": "^1.1.4",
+    "@fortawesome/fontawesome-free-solid": "^5.0.7",
+    "@fortawesome/react-fontawesome": "0.0.17",
```
Comment (marklabs) [inline package.json]: is this needed at this time?

File: src/index.js (hunk 1 above)
Comment (JDWong-wag) [reply src/index.js]: The react-widgets dateTime picker needs it

File: src/containers/Booking/BookingWizard/CCInfo/validate.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    ])(values) };
+
+  const detectedCreditCard = creditCardType(values.cardNumber)[0]; // should only be one possible card type
+  // const securityCodeLength = detectedCreditCard ? detectedCreditCard.code.size : null;
+
+  return {
+    ...errors,
+    cardNumber: validateCardNumber(values.cardNumber, detectedCreditCard),
+  };
+};
```
Comment (marklabs) [inline src/containers/Booking/BookingWizard/CCInfo/validate.js]: very cool validation going on here

File: package.json (hunk 2 above)
Comment (JDWong-wag) [reply package.json]: This enables us to use font-awesome icons.  Currently only being used to display the credit card icon in the payment form

File: src/containers/Booking/BookingWizard/CCInfo/validate.js (hunk 3 above)
Comment (JDWong-wag) [reply src/containers/Booking/BookingWizard/CCInfo/validate.js]: just realized the validator doesn't actually enforce CVC length rules. I'll fix that for the next PR

File: src/components/forms/fields/CheckboxGroup/CheckboxGroup.js [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+      return React.createElement(checkboxTypes[this.props.checkboxType], {
+        key: `checkbox-${label}`,
+        label,
+        inputName,
+        value,
+        checked,
+        onChange: handleChange,
+        onFocus,
+      });
+    });
```
Comment (marklabs) [inline src/components/forms/fields/CheckboxGroup/CheckboxGroup.js]: just giving notice to the very clean code style of use of higher-order functions!

**Pull request merged**